osbenchmark/metrics.py

```diff
@@ -1315,6 +1315,45 @@ def format_dict(d):
         console.println("")
         console.println("No recent test_executions found.")
 
+def list_aggregated_test_results(cfg):
+    def format_dict(d):
+        if d:
+            items = sorted(d.items())
+            return ", ".join(["%s=%s" % (k, v) for k, v in items])
+        else:
+            return None
+
+    aggregated_test_executions = []
+    for test_execution in test_execution_store(cfg).list_aggregations():
+        aggregated_test_executions.append([
+            test_execution.test_execution_id,
+            time.to_iso8601(test_execution.test_execution_timestamp),
+            test_execution.workload,
+            format_dict(test_execution.workload_params),
+            test_execution.test_procedure_name,
+            test_execution.provision_config_instance_name,
+            format_dict(test_execution.user_tags),
+            test_execution.workload_revision,
+            test_execution.provision_config_revision])
+
+    if len(aggregated_test_executions) > 0:
+        console.println("\nRecent aggregated test executions:\n")
+        console.println(tabulate.tabulate(
+            aggregated_test_executions,
+            headers=[
+                "TestExecution ID",
+                "TestExecution Timestamp",
+                "Workload",
+                "Workload Parameters",
+                "TestProcedure",
+                "ProvisionConfigInstance",
+                "User Tags",
+                "workload Revision",
+                "Provision Config Revision"
+                ]))
+    else:
+        console.println("")
+        console.println("No recent aggregate tests found.")
```

Comment:
This works but a lot of this is duplicated from `list_test_executions()`. How do you feel about creating a separate helper method and just passing in the methods like the following pseudo code"

```
def list_test_executions(cfg):
    list_test_helper(test_execution_store(cfg).list_aggregations(), "test executions")

def list_aggregated_test_results(cfg):
    list_test_helper(test_execution_store(cfg).list_aggregations(), "aggregation tests")

def list_test_helper(test_store_method, test_execution_type):
    def format_dict(d):
        if d:
            items = sorted(d.items())
            return ", ".join(["%s=%s" % (k, v) for k, v in items])
        else:
            return None
    
    test_executions = []
    listed_test_executions = test_store_method
    for test_execution in listed_test_executions:
        ...
    console.println(f"No recent {test_execution_type} found.")
     ...
```
By putting boiler plate code in `list_test_helper()`, we will avoid repeating code and slim down both `list_test_executions(cfg)` and `list_aggregated_test_results(cfg)`.

Reply:
Good point! Did this